stable/democratic-csi/values.yaml

```diff
@@ -296,3 +296,5 @@ volumeSnapshotClasses: []
 #  secrets:
 #    snapshotter-secret:
 
+# Force namespace of namespaced resources
```

Comment:
Let's rename this `namespaceOverride` and put the parameter just below `fullnameOverride: ""` and give it a default value of `""`.

Reply:
Thanks for the review !

Upon the compilation of the chart, everything seems to be updated fine in the namespace I defined (storage):

```
compiled/storage/manifests/storage/democratic-csi/templates/controller-rbac.yaml:10:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/controller-rbac.yaml:255:    namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:11:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:32:            - --leader-election-namespace=storage
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:47:                  fieldPath: metadata.namespace
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:61:            - --leader-election-namespace=storage
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:75:                  fieldPath: metadata.namespace
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:89:            - --leader-election-namespace=storage
compiled/storage/manifests/storage/democratic-csi/templates/controller.yaml:103:                  fieldPath: metadata.namespace
compiled/storage/manifests/storage/democratic-csi/templates/node-rbac.yaml:10:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/node-rbac.yaml:73:    namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/configmap.yaml:12:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/node.yaml:11:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/driver-config-secret.yaml:10:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:5:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:13:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:21:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:29:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:37:  namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:58:  csi.storage.k8s.io/controller-expand-secret-namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:60:  csi.storage.k8s.io/controller-publish-secret-namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:62:  csi.storage.k8s.io/node-publish-secret-namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:64:  csi.storage.k8s.io/node-stage-secret-namespace: storage
compiled/storage/manifests/storage/democratic-csi/templates/storage-classes.yaml:66:  csi.storage.k8s.io/provisioner-secret-namespace: storage
```

Reply:
Was that rendered using the `namespaceOverride` value set? I'm relatively confident we still don't have all the changes needed to support that param.

Reply:
In short any places the directly reference `Release.Namespace` won't work 100% with the `namespaceOverride` parameter. For example:

```
  csi.storage.k8s.io/{{ $k }}-namespace: {{ $root.Release.Namespace }}
```

All the other references in the `egrep` output above will need to be reworked as well. Otherwise we should remove the helper and just use `Release.Namespace` everywhere and not support `namespaceOverride` at all.

Reply:
Oh yes! You are right indeed!

Than it will be too much haasle to support this and we should maybe only rely on `.Release.Namespace`.

I'll have a look at this tomorrow.

Thanks for your inputs 😉

Reply:
I think it probably makes a lot of sense to stick with standard `.Release.Namespace` semantics yes.

Reply:
Done!

Reply:
Just a couple changes then we should be good.

- place namespace immediately following name wherever possible 
- I think we should use $root in the snapshot classes

Reply:
I think we should be good to go now :)

Thanks for your review!

Reply:
Looks good!